modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -545,7 +545,7 @@ Mat getMatFromTensor(opencv_onnx::TensorProto& tensor_proto)
         CV_Error(Error::StsUnsupportedFormat, "Unsupported data type: " +
                         opencv_onnx::TensorProto_DataType_Name(datatype));
     if (tensor_proto.dims_size() == 0)
-        blob.dims = 1;  // To force 1-dimensional cv::Mat for scalars.
+        CV_CheckEQ(blob.dims, 1, "Tensor is scalar");  // ensure 1-dimensional cv::Mat for scalars
```

Comment:
Current list of failed DNN tests require better and accurate handing of shapes in ONNX importer and probably in layers implementations.

We should use 0D tensor (scalar) here. 1D doesn't work properly.